eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -418,21 +420,23 @@ public synchronized BackupRegistry get() {
         } catch (Throwable e) {
             throw new RuntimeException("Failed to initialize DiscoveryClient!", e);
         }
-        if (clientConfig.shouldFetchRegistry() && !fetchRegistry(false)) {
-            fetchRegistryFromBackup();
-        }
 
         // Configure new transport layer (candidate for injecting in the future)
+        EurekaHttpClientFactory newEurekaHttpClientFactory = null;
         EurekaHttpClient newEurekaHttpClient = null;
         try {
-            String[] availZones = clientConfig.getAvailabilityZones(clientConfig.getRegion());
-            String myZone = InstanceInfo.getZone(availZones, instanceInfo);
-            newEurekaHttpClient = EurekaHttpClients.createStandardClient(clientConfig, applicationInfoManager, myZone);
+            newEurekaHttpClientFactory = EurekaHttpClients.createStandardClientFactory(clientConfig, applicationInfoManager);
```

Comment:
Before that we need to get rid of legacy transport related code in DiscoveryClient. We want to keep it that way for some time, until we have enough confidence in the new transport implementation.